StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1311CodeFixProvider.cs

```diff
@@ -43,8 +47,58 @@ public override ImmutableArray<string> GetFixableDiagnosticIds()
                     continue;
 
                 var newName = char.ToUpper(token.ValueText[0]) + token.ValueText.Substring(1);
-                context.RegisterFix(CodeAction.Create($"Rename field to '{newName}'", cancellationToken => RenameHelper.RenameSymbolAsync(document, root, token, newName, cancellationToken)), diagnostic);
+
+                var typeSyntax = this.GetParentTypeDeclaration(token);
+
+                if (typeSyntax != null)
+                {
+                    SemanticModel semanticModel = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+                    var typeDeclarationSymbol = semanticModel.GetDeclaredSymbol(typeSyntax) as INamedTypeSymbol;
+                    if (typeDeclarationSymbol == null)
+                    {
+                        return;
+                    }
+
+                    while (!this.IsValidNewMemberName(typeDeclarationSymbol, newName) || newName == typeDeclarationSymbol.Name)
```

Comment:
@pdelvo As a further clarification, I am not opposed to expanding the algorithm for this code fix in the future, but I think each move in that direction should be narrowly scoped and designed to address a particularly common scenario. For this pull request, the only situation I'm hoping to cover is the simplest case where a backing field and property differ only by the casing of the first letter. :+1: